libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/websocket/WebsocketRouteAdaptor.kt

```diff
@@ -87,7 +100,7 @@ internal class WebsocketRouteAdaptor(
     // The handler is called when an error is detected.
     override fun handleError(ctx: WsErrorContext) {
         try {
-            requireNotNull(channel).onError?.let { it(ctx.error()) }
+            requireNotNull(channels[ctx.sessionId]).onError?.invoke(ctx.error())
```

Comment:
Should we also remove from `channels` map:
```suggestion
            channels.remove(ctx.sessionId)?.onError?.invoke(ctx.error())
```
?

Reply:
Afaik onClose is always called after onError by Jetty, so no need to perform any closing here.